[JSC] Drop support for the softfp ABI on ARMv7

xanlpz opened the pull request:
#### a2ec4ef1997d6fafa6ffc607bffb54e76168a918
<pre>
[JSC] Drop support for the softfp ABI on ARMv7
<a href="https://bugs.webkit.org/show_bug.cgi?id=242172">https://bugs.webkit.org/show_bug.cgi?id=242172</a>

Reviewed by Yusuke Suzuki.

Remove for all purposes the ARM_HARDFP option and just delete all code
enabled when it&apos;s false. It remains used in PlatformEnable.h, so we
enable CLOOP for ARMv7 builds using the softfp ABI.

* Source/JavaScriptCore/assembler/testmasm.cpp:
(JSC::testStoreBaseIndex):
* Source/JavaScriptCore/dfg/DFGSpeculativeJIT.h:
(JSC::DFG::SpeculativeJIT::appendCallSetResult):
* Source/JavaScriptCore/jit/CCallHelpers.h:
(JSC::CCallHelpers::setupArgumentsImpl):
* Source/JavaScriptCore/jit/FPRInfo.h:
(JSC::FPRInfo::toArgumentRegister):
* Source/WTF/wtf/PlatformCPU.h:
* Source/WTF/wtf/PlatformEnable.h:

Canonical link: <a href="https://commits.webkit.org/252034@main">https://commits.webkit.org/252034@main</a>
</pre>

aperezdc reviewed (commented):
@xanlpz It looks like you didn't use `git webkit pfr` to upload the pull request, the description message here does not follow the format generated by the tool.

Other than the nitpick above, I hoped more code could have been removed, but the cleanup is welcome nevertheless, thanks! rs=me, but I'll let the JSC folks give the actual r+ 😸

Constellation reviewed (approved):
r=me

xanlpz committed:
[JSC] Drop support for the softfp ABI on ARMv7
https://bugs.webkit.org/show_bug.cgi?id=242172

Reviewed by Yusuke Suzuki.

Remove for all purposes the ARM_HARDFP option and just delete all code
enabled when it's false. It remains used in PlatformEnable.h, so we
enable CLOOP for ARMv7 builds using the softfp ABI.

* Source/JavaScriptCore/assembler/testmasm.cpp:
(JSC::testStoreBaseIndex):
* Source/JavaScriptCore/dfg/DFGSpeculativeJIT.h:
(JSC::DFG::SpeculativeJIT::appendCallSetResult):
* Source/JavaScriptCore/jit/CCallHelpers.h:
(JSC::CCallHelpers::setupArgumentsImpl):
* Source/JavaScriptCore/jit/FPRInfo.h:
(JSC::FPRInfo::toArgumentRegister):
* Source/WTF/wtf/PlatformCPU.h:
* Source/WTF/wtf/PlatformEnable.h:

Canonical link: https://commits.webkit.org/252034@main
